### PR TITLE
Modernize functional tests III

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: pulp_ansible/tests/functional/api/role/test_crud_distribution.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    assert "Invalid hyperlink - No URL match." in exc_info.value.body
+
+    ansible_distro_api_client.partial_update(distribution.pulp_href, {"repository": None})
+    monitor_task(ansible_distro_api_client.partial_update(distribution.pulp_href, {"repository_version": repository.latest_version_href}).task)
+    distribution = ansible_distro_api_client.read(distribution.pulp_href)
+    assert distribution.repository is None
+    assert distribution.repository_version == repository.latest_version_href
+
+    if False:
+        # TODO This fails by complaining repository and version cannot be used together.
```
Comment (mdellweg) [inline pulp_ansible/tests/functional/api/role/test_crud_distribution.py:35]: This is actually a pulpcore bug.

Comment (mdellweg) [reply pulp_ansible/tests/functional/api/role/test_crud_distribution.py:49]: https://github.com/pulp/pulpcore/pull/4325

File: pulp_ansible/tests/functional/api/role/test_download_content.py [hunk 2]
```diff
@@ -1,97 +1,36 @@
...
+    roles = ansible_roles_api_client.list(repository_version=repository.latest_version_href)
+    role = roles.results[0]
+    unit_path = f"{role.namespace}/{role.name}/{role.version}.tar.gz"
+    pulp_url = urljoin(distribution.base_url, unit_path)
+    pulp_response = requests.get(pulp_url)
+    pulp_response.raise_for_status()
+    pulp_hash = hashlib.sha256(pulp_response.content).hexdigest()
+
+    # TODO That's not how you get roles from galaxy v1...
+    assert pulp_hash is not None
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulp_ansible/tests/functional/api/role/test_download_content.py:31]: Not yet. And also this test was skipped before. So do you think it brings new value, or should we remove it altogether?

File: pulp_ansible/app/serializers.py [hunk 3]
```diff
@@ -358,6 +358,7 @@ class Meta:
             "pulp_href",
             "pulp_created",
             "base_path",
+            "base_url",
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulp_ansible/app/serializers.py:361]: I'm going to remove it again and mark the dysfunctional test mentioned above skipped.
(Maybe we get back to the roles api eventually...)

File: pulp_ansible/tests/functional/api/role/test_download_content.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gerrod3) [reply pulp_ansible/tests/functional/api/role/test_download_content.py:33]: I personally think this test doesn't add much value. We already test that we can download roles in the cli test when we check we can install a role from Pulp. I would suggest removing it.

**Review by gerrod3 — APPROVED** [review]

**Pull request merged**